### PR TITLE
Present ExternalUserAgent without passing UIViewController for iOS 11+

### PR DESCRIPTION
**Pull request by kevinjohnason**
This PR is referring to #338, iOS 11+ devices don't need to pass UIViewController in order to present an external user agent. 

```objc
// builds authentication request for apps that support iOS 11+
OIDAuthorizationRequest *request =
    [[OIDAuthorizationRequest alloc] initWithConfiguration:configuration
                                                  clientId:kClientID
                                                    scopes:@[OIDScopeOpenID,
                                                             OIDScopeProfile]
                                               redirectURL:KRedirectURI
                                              responseType:OIDResponseTypeCode
                                      additionalParameters:nil];

// performs authentication request
AppDelegate *appDelegate =
    (AppDelegate *)[UIApplication sharedApplication].delegate;


appDelegate.currentAuthorizationFlow =
    [OIDAuthState authStateByPresentingAuthorizationRequest:request
                        callback:^(OIDAuthState *_Nullable authState,
                                   NSError *_Nullable error) {
  if (authState) {
    NSLog(@"Got authorization tokens. Access token: %@",
          authState.lastTokenResponse.accessToken);
    [self setAuthState:authState];
  } else {
    NSLog(@"Authorization error: %@", [error localizedDescription]);
    [self setAuthState:nil];
  }
}];
```

### CODE REVIEW

**Comment by WilliamDenniss** [comment]
Can you confirm if this contribution made under the existing [WeWork CLA](https://openid.net/wordpress-content/uploads/2018/06/OIDF_CLA-Agreement-Corporate_WeWork-Companies_2018-06-21.pdf)?

**Comment by kevinjohnason** [comment]
Yes that is our CLA. Do I need to sign anything to link with that?

File: Source/iOS/OIDAuthState+IOS.h [hunk 1]
```diff
@@ -45,6 +45,17 @@ NS_ASSUME_NONNULL_BEGIN
                      presentingViewController:(UIViewController *)presentingViewController
                                      callback:(OIDAuthStateAuthorizationCallback)callback;
 
+/*! @brief Convenience method for iOS 11+ devices to create a @c OIDAuthState by presenting an    authorization request and performing the authorization code exchange in the case of code flow requests. For the hybrid flow, the caller should validate the id_token and c_hash, then perform the token request (@c OIDAuthorizationService.performTokenRequest:callback:) and update the OIDAuthState with the results (@c OIDAuthState.updateWithTokenResponse:error:).
```
Comment (WilliamDenniss) [inline Source/iOS/OIDAuthState+IOS.h]: Can you add some linebreaks here?

File: Source/iOS/OIDAuthState+IOS.m [hunk 2]
```diff
@@ -34,4 +34,12 @@ @implementation OIDAuthState (IOS)
                                                 callback:callback];
 }
 
++ (id<OIDExternalUserAgentSession>)
+    authStateByPresentingAuthorizationRequest:(OIDAuthorizationRequest *)authorizationRequest
+                                     callback:(OIDAuthStateAuthorizationCallback)callback {
+  return [self authStateByPresentingAuthorizationRequest:authorizationRequest
+                                presentingViewController:[UIViewController new]
```
Comment (WilliamDenniss) [inline Source/iOS/OIDAuthState+IOS.m]: I don't like the use of `[UIViewController new]` as it bypasses the intent here.

Instead I think there should be a constructor for OIDExternalUserAgentIOS that doesn't require a UIViewController, and that is what should be used.

**Comment by WilliamDenniss** [comment]
Hi @kevinjohnason, I'm supportive of this change (once the comments are addressed), do we want to try and get it in?

**Comment by kevinjohnason** [comment]
yes, I'll get update this PR before end of this week. :)

**Comment by kevinjohnason** [comment]
hey @WilliamDenniss, I simply made OIDExternalUserAgentIOS not requiring UIViewController as instructed. 

I was thinking maybe there can be a subclass of OIDExternalUserAgentIOS that does require UIViewController and have the logic below iOS 11 moved there. Do you see this fitting in the PR?

**Comment by WilliamDenniss** [comment]
> I was thinking maybe there can be a subclass of OIDExternalUserAgentIOS that does require UIViewController and have the logic below iOS 11 moved there. Do you see this fitting in the PR?

I think a separate PR, if any. What would be the benefit? Just to separate the old code?

File: Source/iOS/OIDExternalUserAgentIOS.m [hunk 3]
```diff
@@ -43,8 +43,12 @@ @implementation OIDExternalUserAgentIOS {
 #pragma clang diagnostic pop
 }
 
+- (nullable instancetype)init {
+  return [self initWithPresentingViewController: nil];
```
[the remark this answers is not in the document]
Comment (kevinjohnason) [reply Source/iOS/OIDExternalUserAgentIOS.m]: sorry about that. habit from swift

File: Source/iOS/OIDAuthState+IOS.h [hunk 4]
```diff
@@ -45,6 +45,22 @@ NS_ASSUME_NONNULL_BEGIN
                      presentingViewController:(UIViewController *)presentingViewController
                                      callback:(OIDAuthStateAuthorizationCallback)callback;
```
[the remark this answers is not in the document]
Comment (kevinjohnason) [reply Source/iOS/OIDAuthState+IOS.h]: yes I agree this method should be deprecated for iOS 11+. Making changes.

**Comment by kevinjohnason** [comment]
> > I was thinking maybe there can be a subclass of OIDExternalUserAgentIOS that does require UIViewController and have the logic below iOS 11 moved there. Do you see this fitting in the PR?
> 
> I think a separate PR, if any. What would be the benefit? Just to separate the old code?

Just cleaner I think.

**Comment by WilliamDenniss** [comment]
Merged. Thanks for your contribution!

**Pull request merged**